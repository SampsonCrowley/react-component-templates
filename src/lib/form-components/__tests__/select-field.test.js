import React from 'react'
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';

import SelectField from 'form-components/select-field'

describe('Form Components - SelectField', () => {
  const div = document.createElement('div');
  const defaultProps = {
    id: 'testId',
    name: 'testName',
    label: 'test label',
    value: 'test value',
    feedback: 'test feedback'
  }

  const createSelectField = ({...props}) => {
    ReactDOM.render((
      <Router>
        <SelectField {...props} />
      </Router>
    ), div);
    return div
  }

  const getSelectFieldInput = ({...props}) => createSelectField(props).querySelector('input')
  const getSelectFieldLabel = ({...props}) => createSelectField(props).querySelector('label')
  const getSelectFieldFeedback = ({...props}) => createSelectField(props).querySelector('small')

  it('renders an input tag', () => {
    const rendered = getSelectFieldInput(defaultProps)
    expect(rendered).toBeTruthy()
    expect(rendered.tagName).toBe("INPUT")
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders an label tag', () => {
    const rendered = getSelectFieldLabel(defaultProps)
    expect(rendered).toBeTruthy()
    expect(rendered.tagName).toBe("LABEL")
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders a feedback tag', () => {
    const rendered = getSelectFieldFeedback(defaultProps)
    expect(rendered).toBeTruthy()
    expect(rendered.tagName).toBe("SMALL")
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders a only an input if skipExtras is passed', () => {
    const feedback = getSelectFieldFeedback({...defaultProps, skipExtras: true})
    const label = getSelectFieldLabel({...defaultProps, skipExtras: true})
    const input = getSelectFieldInput({...defaultProps, skipExtras: true})
    expect(feedback).toBe(null)
    expect(label).toBe(null)
    expect(input).toBeTruthy()
    expect(input.tagName).toBe('INPUT')
    ReactDOM.unmountComponentAtNode(div);
  })

  it('is snapshotable', () => {
    const tree = renderer
      .create(
        <Router>
          <SelectField {...defaultProps}/>
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot()
  })
})
