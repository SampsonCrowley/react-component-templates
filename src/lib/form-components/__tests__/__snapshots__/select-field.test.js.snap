// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Components - SelectField is snapshotable 1`] = `
Array [
  <label
    htmlFor="testId"
  >
    test label
  </label>,
  <style>
    input[id^="react-select-"] { opacity: 1 !important; }
  </style>,
  <input
    autoComplete="off"
    id="testId"
    name="testName"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onKeyUp={[Function]}
    style={
      Object {
        "border": "none",
        "boxShadow": 0,
        "display": "block",
        "height": 0,
        "visibility": "",
        "width": 0,
      }
    }
    tabIndex={-1}
    type="text"
    value="test value"
  />,
  <div
    className="text-dark css-2b097c-container"
    id="testId.select"
    onKeyDown={[Function]}
  >
    <div
      className=" css-yk16xz-control"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className=" css-l1xr2i-Component"
      >
        <div
          className=" css-1wa3eu0-placeholder"
        >
          Select...
        </div>
        <div
          className="css-xry0bu-Component"
        >
          <div
            className=""
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="react-select-6-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "label": "input",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className=" css-ny0e4k-Component"
      >
        <span
          className=" css-1okebmr-indicatorSeparator"
        />
        <div
          aria-hidden="true"
          className=" css-tlfecz-indicatorContainer"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg
            aria-hidden="true"
            className="css-7ezgaa-Component"
            focusable="false"
            height={20}
            viewBox="0 0 20 20"
            width={20}
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  <small
    className="form-control-focused"
  >
    test feedback
  </small>,
  <span
    tabIndex="0"
  />,
]
`;
